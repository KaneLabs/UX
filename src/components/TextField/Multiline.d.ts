declare const Multiline: ({ mobile, flat, style, autoFocus, onChangeText, ...props }: {
    [x: string]: any;
    mobile?: boolean | undefined;
    flat?: boolean | undefined;
    style: any;
    autoFocus?: boolean | undefined;
    onChangeText?: ((text: string) => null) | undefined;
}) => JSX.Element;
export default Multiline;
