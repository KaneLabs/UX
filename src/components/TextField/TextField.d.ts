import React from 'react';
declare const TextField: React.ForwardRefExoticComponent<React.RefAttributes<unknown>>;
export default TextField;
