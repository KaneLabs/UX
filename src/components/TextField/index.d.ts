export { default as Multiline } from './Multiline';
export { default, default as TextField } from './TextField';
