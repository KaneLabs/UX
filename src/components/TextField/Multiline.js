import React, { useState } from 'react';
import { StyleSheet, TextInput } from 'react-native';
// import { AccountAvatar } from 'eros-ui/components';
import { makeStyles } from 'eros-ui/theme';

const Multiline = ({
  mobile = false,
  flat = false,
  style,
  autoFocus = false,
  onChangeText = () => null,
  ...props
}) => {
  const [height, setHeight] = useState('auto');
  const [focused, setFocused] = useState(autoFocus);

  const onContentSizeChange = (event) => {
    const { height } = event.nativeEvent.contentSize;
    setHeight(height);
  };

  const onChange = (nextText) => {
    if (nextText.length === 0) {
      setHeight('auto');
    }
    return onChangeText(nextText);
  };

  return (
    <TextInput
      multiline
      blurOnSubmit={false}
      placeholder="What's up?"
      scrollEnabled={false}
      onFocus={() => setFocused(true)}
      onBlur={() => setFocused(false)}
      onChangeText={onChange}
      onContentSizeChange={onContentSizeChange}
      style={[
        styles.multiline,
        mobile ? styles.mobile : styles.desktop,
        { height },
        flat && styles.flat,
        focused && !flat && styles.focused,
        style,
      ]}
      {...props}
    />
  );
};

const styles = makeStyles((theme) => ({
  multiline: {
    width: '100%',
    padding: unit,
    color: textColor.primary,
    borderRadius,
    borderColor: 'transparent',
    borderWidth,
  },
  flat: {
    borderWidth: 0,
  },
  focused: {
    borderColor: primaryColorOpaque,
  },
  mobile: {
    fontSize: 18,
    lineHeight: 27,
    fontWeight: '300',
  },
  desktop: {
    fontSize: 21,
    lineHeight: 32,
    fontWeight: '300',
  },
}));

export default Multiline;
