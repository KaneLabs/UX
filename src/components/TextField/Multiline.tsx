import React, { useState } from 'react';
import { TextInput, StyleProp } from 'react-native';
// import { AccountAvatar } from 'eros-ui/components';
import { makeStyles, Theme } from 'eros-ui/theme';

const Multiline = ({
  mobile = false,
  flat = false,
  style,
  autoFocus = false,
  onChangeText = (text: string) => null,
  ...props
}) => {
  const [height, setHeight] = useState('auto');
  const [focused, setFocused] = useState(autoFocus);

  const onContentSizeChange = (event: any) => {
    const { height } = event.nativeEvent.contentSize;
    setHeight(height);
  };

  const onChange = (nextText: string) => {
    if (nextText.length === 0) {
      setHeight('auto');
    }
    return onChangeText(nextText);
  };

  return (
    <TextInput
      multiline
      blurOnSubmit={false}
      placeholder="What's up?"
      scrollEnabled={false}
      onFocus={() => setFocused(true)}
      onBlur={() => setFocused(false)}
      onChangeText={onChange}
      onContentSizeChange={onContentSizeChange}
      style={[
        styles.multiline,
        mobile ? styles.mobile : styles.desktop,
        { height },
        flat && styles.flat,
        focused && !flat && styles.focused,
        style,
      ]}
      {...props}
    />
  );
};

const styles = makeStyles((theme: Theme) => ({
  multiline: {
    width: '100%',
    padding: theme.unit,
    color: theme.textColor.primary,
    borderRadius: theme.borderRadius,
    borderColor: 'transparent',
    borderWidth: theme.borderWidth,
  },
  flat: {
    borderWidth: 0,
  },
  focused: {
    borderColor: theme.primaryColorOpaque,
  },
  mobile: {
    fontSize: 18,
    lineHeight: 27,
    fontWeight: '300',
  },
  desktop: {
    fontSize: 21,
    lineHeight: 32,
    fontWeight: '300',
  },
}));

export default Multiline;
