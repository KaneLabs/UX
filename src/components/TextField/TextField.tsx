import * as React from 'react';
import {
  StyleSheet,
  TextInput,
  Platform,
  TextInputProps,
  ViewStyle,
} from 'react-native';

import { useTheme, makeStyles, Theme, ThemeModes } from 'eros-ui/theme';

interface TextFieldProps extends TextInputProps {
  flat?: boolean;
  hoverStyle?: ViewStyle;
  focusStyle?: ViewStyle;
  children?: React.ReactNode;
}

const TextField = React.forwardRef<TextInput, TextFieldProps>((props, ref) => {
  const {
    autoFocus = false,
    style = null,
    flat = false,
    hoverStyle,
    focusStyle,
    ...rest
  }: TextFieldProps = props;
  const [theme] = useTheme();
  const [focus, setFocus] = React.useState(autoFocus);
  const [hover, setHover] = React.useState(false);
  const styles = useStyles();

  const textFieldStyle = StyleSheet.flatten([
    styles.default,
    hover && StyleSheet.compose(styles.hover, hoverStyle),
    focus && StyleSheet.compose(styles.focus, focusStyle),
    flat && styles.flat,
    Platform.OS === 'web' && { outlineWidth: 0 }, // needed for Web
    style,
  ]);

  const keyboardAppearance = theme.mode === ThemeModes.light ? 'light' : 'dark';
  return (
    <TextInput
      ref={ref}
      style={textFieldStyle}
      onFocus={() => setFocus(true)}
      onBlur={() => setFocus(false)}
      keyboardAppearance={keyboardAppearance}
      placeholderTextColor={
        focus || hover ? theme.textColor.secondary : theme.textColor.disabled
      }
      {...rest}
    />
  );
});

const useStyles = makeStyles((theme: Theme) => ({
  default: Platform.select({
    default: {
      fontSize: 16,
      lineHeight: 19,
      borderRadius: theme.unit,
      borderColor: theme.borderColor,
      borderWidth: theme.borderWidth,
      color: theme.textColor.secondary,
      alignItems: 'center',
      paddingLeft: theme.unit,
      paddingRight: theme.unit,
      paddingTop: theme.unit / 2,
      paddingBottom: theme.unit / 2,
    },
  }),
  flat: Platform.select({
    web: { borderWidth: 0, outlineWidth: 0 },
    default: { borderWidth: 0 },
  }),
  hover: {
    borderColor: theme.textColor.disabled,
    color: theme.textColor.secondary,
    ...theme.shadow(8),
    backgroundColor: theme.canvas2Opaque,
  },
  focus: {
    borderColor: theme.textColor.disabled,
    color: theme.textColor.secondary,
    ...theme.shadow(8),
    backgroundColor: theme.canvas2Opaque,
  },
}));

export default TextField;
