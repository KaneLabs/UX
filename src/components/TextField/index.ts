export { default, default as TextField } from './TextField';
export { default as Multiline } from './Multiline';
