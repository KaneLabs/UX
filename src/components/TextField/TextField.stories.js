'use strict';
var __importDefault =
  (this && this.__importDefault) ||
  function (mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
const react_1 = __importDefault(require('react'));
const react_native_1 = require('react-native');
const addon_knobs_1 = require('@storybook/addon-knobs');
const Container_1 = __importDefault(
  require('eros-ui-storybook/components/Container'),
);
const storiesOf_1 = require('eros-ui-storybook/helpers/storiesOf');
const TextField_1 = __importDefault(require('./TextField'));
exports.default = storiesOf_1
  .storiesOf('Components|TextField', module)
  .addParameters({ jest: ['TextField'] })
  .add('TextField', () => (
    <Container_1.default>
      <react_native_1.View
        style={{ flexDirection: 'column', flexWrap: 'wrap' }}>
        <TextField_1.default
          placeholder="Enter Text"
          value={addon_knobs_1.text(
            'Text',
            'The spectacle before us was indeed sublime.',
          )}
        />
      </react_native_1.View>
    </Container_1.default>
  ));
