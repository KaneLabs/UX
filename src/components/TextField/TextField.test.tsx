import React from 'react';
import renderer from 'react-test-renderer';
import TextField from './TextField';
import ThemeProvider from '../../theme/ThemeProvider';

test('TextField Renders', () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <TextField placeholder="Text Field" />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
