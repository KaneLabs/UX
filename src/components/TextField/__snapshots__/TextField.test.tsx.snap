// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Renders 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardAppearance="light"
  onBlur={[Function]}
  onFocus={[Function]}
  placeholder="Text Field"
  placeholderTextColor="rgba(0,0,0,0.38)"
  rejectResponderTermination={true}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "rgba(0,0,0,0.12)",
      "borderRadius": 8,
      "borderWidth": 0.5,
      "color": "rgba(0,0,0,0.54)",
      "fontSize": 16,
      "lineHeight": 19,
      "paddingBottom": 4,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 4,
    }
  }
  underlineColorAndroid="transparent"
/>
`;
